File: frontend/types/usage.ts
export interface UsageSummary {
  user_id: string
  period: {
    start: string
    end: string
  }
  totals: {
    operations: number
    tokens: number
    cost: number
  }
  by_operation: Record<string, {
    count: number
    tokens: number
    cost: number
  }>
}

export interface DailyUsage {
  date: string
  tokens: number
  cost: number
  operations: number
}

File: backend/schemas/usage.py
"""Pydantic schemas for usage tracking."""

from pydantic import BaseModel
from typing import Dict, Any, List


class UsageSummaryResponse(BaseModel):
    user_id: str
    period: Dict[str, str]  # {start, end}
    totals: Dict[str, Any]  # {operations, tokens, cost}
    by_operation: Dict[str, Dict[str, Any]]  # {operation_name: {count, tokens, cost}}


class DailyUsageEntry(BaseModel):
    date: str
    tokens: int
    cost: float
    operations: int


class DailyUsageResponse(BaseModel):
    daily_usage: List[DailyUsageEntry]
